Makefile

```diff
@@ -1,52 +1,191 @@
+CONFIG_APP_CODE        += ./cmd/rtcd
+# Dockerfile's location
+CONFIG_DOCKERFILE      += ./build/Dockerfile
+# Docker options to inherit for all docker run commands
+CONFIG_DOCKEROPTS      += --rm -i -u $$(id -u):$$(id -g) --platform "linux/amd64"
+# Registry to upload images
+CONFIG_IMAGE_REGISTRY  += my.registy.com
+CONFIG_DOCKER_REPO     += mattermost
+# Temporary folder to output compiled binaries artifacts
+CONFIG_BIN_DIR         := dist
+## Docker Images Variables
+IMAGE_GOLINT           += "golangci/golangci-lint:v1.45.2@sha256:e84b639c061c8888be91939c78dae9b1525359954e405ab0d9868a46861bd21b"
+IMAGE_DOCKERLINT       += "hadolint/hadolint:v2.9.2@sha256:d355bd7df747a0f124f3b5e7b21e9dafd0cb19732a276f901f0fdee243ec1f3b"
+
+## Go Variables
+# Go executable
+GO         = go
+# LDFLAGS
+GO_LDFLAGS += -X "github.com/mattermost/${APP_NAME}/service.buildHash=$(APP_COMMIT)"
+#GO_LDFLAGS += -X "github.com/mattermost/${APP_NAME}/[...]])"
+# Architectures to build for
+GO_BUILD_PLATFORMS=linux-amd64 linux-arm64 darwin-amd64 darwin-arm64
+GO_BUILD_PLATFORMS_ARTIFACTS=$(foreach cmd,$(addprefix go-build/,${APP_NAME}),$(addprefix $(cmd)-,$(GO_BUILD_PLATFORMS)))
+# Build options
+GO_BUILD_OPTS += -mod=readonly -trimpath
+GO_TEST_OPTS += -mod=readonly -failfast -race
+
+## General Variables
+# Use repostiroy name as application name
+APP_NAME    = $(shell basename -s .git `git config --get remote.origin.url`)
+# Get current commit
+APP_COMMIT  = $(shell git log --pretty=format:'%h' -n 1)
+# Check if we are in a release tag, if yes use it as app version.
+APP_VERSION = $(shell git describe --abbrev=0 --exact-match --tags > /dev/null 2>&1 || echo dev-$(APP_COMMIT))
+BUILD_DATE  = $(shell date -u)
+
+## General Configuration Variables
+# We don't need make's built-in rules.
+MAKEFLAGS += --no-builtin-rules
+# Be pedantic about undefined variables.
+MAKEFLAGS += --warn-undefined-variables
+# Set help as default target
+.DEFAULT_GOAL := help
+
+# ====================================================================================
+# Colors
+
+BLUE         := $(shell printf "\033[34m")
+YELLOW       := $(shell printf "\033[33m")
+RED          := $(shell printf "\033[31m")
+GREEN        := $(shell printf "\033[32m")
+CNone        := $(shell printf "\033[0m")
+
+# ====================================================================================
+# Logger
+
+TIME_LONG	= `date +%Y-%m-%d' '%H:%M:%S`
+TIME_SHORT	= `date +%H:%M:%S`
+TIME		= $(TIME_SHORT)
+
+INFO	= echo ${TIME} ${BLUE}[ .. ]${CNone}
+WARN	= echo ${TIME} ${YELLOW}[WARN]${CNone}
+ERR		= echo ${TIME} ${RED}[FAIL]${CNone}
+OK		= echo ${TIME} ${GREEN}[ OK ]${CNone}
+FAIL	= (echo ${TIME} ${RED}[FAIL]${CNone} && false)
+
+# ====================================================================================
+# Targets
+
+help: ## to get help
+	@echo "Usage:"
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) |\
+	awk 'BEGIN {FS = ":.*?## "}; {printf "make \033[36m%-30s\033[0m %s\n", $$1, $$2}'
```

Comment:
I think the part here we need to consider is that some people may haven't installed `awk` ever. if we have in mind that in parallel we want them to install less tooling (eg. move to docker) from now on we need to think how we can change this.

Reply:
Yes, I get the need for decoupling from tooling installed locally in users. 
This is the reason behind  designing `docker-lint` and `go-lint` to not depend on local tooling installed. 

Not sure if AWK is part of GNUtils, but I am pretty sure it is part of most -UNIX OSes. MacOS and Linux have it. 

Other than that, the only purpose for this target, is to dynamically format the `makefile help`, thus to produce this output: 
```
make help
Usage:
make help                           to get help
make lint                           to lint all
make test                           to test all
make docker-build                   to build the docker image
make docker-push                    to push the docker image
make docker-sbom                    to print a sbom report
make docker-scan                    to print a vulnerability report
make docker-lint                    to lint the Dockerfile
make go-build                       to build binaries
make go-run                         to run locally for development
make go-test                        to run tests
make go-mod-check                   to check go mod files consistency
make go-update-dependencies         to update go dependencies (vendor)
make go-lint                        to lint go code
make go-doc                         to generate documentation
make clean                          to clean-up
```

From my point of view:
- Lunching dockerd
- Waiting for linux image to download (containing awk + make) 
- Running the linux container mounting the local volume containing the makefile
- Running `make help` , to produce the output

Will introduce some delay in showing the user the actual makefile help. 

I guess we could format it under a docker container, Would you vote in favor of it?

Reply:
+1 🎸